### DISCUSSION TITLE
Can you use this to select, in one shot, a particular 'projector' (Win+P) setting? e.g. "PC screen only" or "Extend"

### DISCUSSION THREAD

**Discussion by RandomNerd1** [ThioJoe/Windows-Super-God-Mode#23] [Q&A]
My ultimate goal is a 'native' method to toggle my second monitor on ("Extend") and off ("PC screen only") in one step, instead of having to press Win+P...have the little thing pop up on the side...then go and select the one I want with the mouse...and then it changes.

I tried a couple of searches through the links it created and couldn't find it, but figured I might just not know how it's named.

**Comment by ThioJoe** [comment]
Turns out there's a tool built into windows in system32 called `DisplaySwitch.exe` which does what you want. You should be able to just open command prompt anywhere to try it out. 

You use it with one of these arguments:
/internal - PC screen only
/external - Second screen only
/extend - Extend
/clone - Duplicate

For example: `displayswitch /external`

Or with numbers I think in Windows 11, though both might work:
1 = /internal
2 = /clone  
3 =  /extend
4 =  /external

So for example: `displayswitch 1`

In your case you could just create two separate shortcuts, one to `displayswitch /internal` and one to `displayswitch /extend`

**Comment by RandomNerd1** [comment]
I probably should have mentioned that I'm already aware of displayswitch. It's _better_ than the Win+P route, but it still pops up a (somewhat different than the Win+P one, for some reason) UI element on the right side before doing its thing, which is not only a bit annoying visually, but also adds a delay to the actual action being taken.

The closest thing I've achieved to what I want involves using this [old, apparently-dead program](https://sourceforge.net/projects/monitorswitcher/), setting up a "profile" with one of my monitors on, another with both of them on, and using it to 'toggle' between them. It's alright, but I was hoping for a more 'raw' way of doing it, if that makes sense. And, ideally, not get that bit of lag/stutter that occurs whenever the 'mode' is changed (which I get no matter which method I use, 'normal', displayswitch, or that 3rd-party app, so maybe it just can't be gotten around, and is a limitation of the OS).

I realize this is a very specific thing I'm trying to do, lol.

**Reply by ThioJoe** [reply]
Hm interesting. When I use displayswitch (using either the /slash version of the command or numbers), it always immediately takes effect. It only pops up the GUI box if I enter it with no arguments.

What version of Windows are you on, and have you tried both types of arguments, slashes and numbers?

**Comment by RandomNerd1** [comment]
I'm on Windows 10.

When using displayswitch, there is no difference between slashes and numbers (e.g. /extend vs. 3).

I did notice that when I use displayswitch, the appearance of the 'popup' on the right side is different. When I press Win+P, it looks like this, as it waits for me to select an option:

https://i.imgur.com/0zR6Skw.png

When I use displayswitch, it looks like this instead (apparently using my accent color as its main color, for some reason), popping up and going away on its own, almost as if the functionality of displayswitch involves invoking that panel and then simulating a click onto one of the options, which ITSELF is the action that actually makes the change, instead of displayswitch doing it itself (if that's true, my thinking is that I want to go to the 'source' and invoke whatever displayswitch is invoking, so that there's no popup on the right, and no delay):

https://imgur.com/INY1wJn

Oh, and just to throw one more fun little wrinkle into this, when _I_ run displayswitch with no arguments, the GUI box that pops up is the FIRST one I linked, not the second--in other words, not the one that appears briefly when I run displayswitch with an argument. In other other words, displayswitch with no argument acts exactly the same as pressing Win+P does.

What the hell Windows, lol...

EDIT: I've noticed that my 'pink version' (might as well call it that lol) of the display mode selector...thing...does come up when I press Win+P, but ONLY if I press Win+P while on the lock screen, at the screen where you're prompted for your password to unlock the PC. Anytime else, Win+P produces the 'dark version' I linked to first...

**Reply by ThioJoe** [reply]
I'll upload the version of DisplaySwitch.exe I have in my windows directory if you want to try it:
[DisplaySwitch.zip](https://github.com/user-attachments/files/16806205/DisplaySwitch.zip)

**Comment by RandomNerd1** [comment]
How about that? Yeah, I get no 'pink version' when I use a shortcut to your displayswitch.exe . I couldn't overwrite my own with it though, got a weird permission error even after I proceeded as administrator. Probably some extra security thing within system32.

But I think I can utilize this anyway, put the .exe wherever and make shortcuts to it, which I use. I'll tinker with this today, thanks. :)

**Comment by RandomNerd1** [comment]
After debating for a bit whether to earnestly try and use some combination of 'blacking out' my second monitor without actually turning it off with [one app](http://multiscreenblank.nookkin.com/) while 'trapping' my mouse cursor within my first monitor (so I don't 'lose' it, lol) with [another](https://dualmonitortool.sourceforge.net/), and using some sort of script/macro to toggle those effects on/off at the same time...I think I've decided to just deal with the little delay that comes with disabling/enabling the second monitor, lol.

Your shared displayswitch helped a lot, thanks! :)
